cartridge/argparse.lua

```diff
@@ -440,46 +440,59 @@ local function get_opts(opts)
 
     local ret = {}
     for optname, opttype in pairs(opts) do
+        local opttype_str = false
+        local opttype_num = false
+        local opttype_bool = false
+        local multi_types = string.gsub(opttype, ' ', '')
+        for _opttype in string.gmatch(multi_types, "[^|]+") do
+            if _opttype == 'string' then
+                opttype_str = true
+            elseif _opttype == 'number' then
+                opttype_num = true
+            elseif _opttype == 'boolean' then
+                opttype_bool = true
+            else
+                return nil, TypeCastError:new(
+                    "can't typecast %s to %s (unsupported type)",
+                    optname, _opttype
+                )
+            end
+        end
+
         local value = args[optname]
         if value == nil then -- luacheck: ignore 542
             -- ignore
         elseif type(value) == opttype then
             ret[optname] = value
-        elseif type(value) == 'number' and opttype == 'string' then
-            ret[optname] = tostring(value)
+        elseif type(value) == 'number' and ( opttype_num or opttype_str ) then
+            local _value = tostring(value)
```

Comment:
I think there it can be just an number. But some tests will be fail.
```
/Users/andrei.sidorov/cartridge/test/unit/argparse_test.lua:331: expected: 
{
    listen = "13301",
}
actual: 
{
    listen = 13301,
}
```

Reply:
Here we could simply fix a test.
I guess 13301 (as number) is valid value for listening.

Reply:
Replace listen to string|number then.
Tarantool accepts numeric listen - https://github.com/tarantool/tarantool/blob/4e9a94a38c381cefc71ae53ca9a959bcdcf6434b/src/box/lua/load_cfg.lua#L167